File: 1-app-basics/NoteKeeper/app/src/main/java/com/thedancercodes/notekeeper/NoteListActivity.java
package com.thedancercodes.notekeeper;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.List;

public class NoteListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_note_list);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        // Populate the ListView
        initializeDisplayContent();
    }

    private void initializeDisplayContent() {

        // Reference to ListView
        ListView listNotes = findViewById(R.id.list_notes);

        // Get list of notes to add into the ListView
        List<NoteInfo> notes = DataManager.getInstance().getNotes();

        // Add the notes into ListView using an ArrayAdapter
        ArrayAdapter<NoteInfo> adapterNotes =
                new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, notes);

        // Associate our Adapter with the ListView
        listNotes.setAdapter(adapterNotes);

    }

}

File: Kotlin/NoteKeeper/1-build-your-first-app/NoteKeeper/app/src/main/java/com/thedancercodes/notekeeper/NoteListActivity.kt
package com.thedancercodes.notekeeper

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.widget.ArrayAdapter

import kotlinx.android.synthetic.main.activity_note_list.*
import kotlinx.android.synthetic.main.content_note_list.*

class NoteListActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_note_list)
        setSupportActionBar(toolbar)

        // Launch Activity for editing notes
        fab.setOnClickListener { view ->
            val activityIntent = Intent(this, MainActivity::class.java)
            startActivity(activityIntent)
        }

        // Populate the ListView using the Adapter
        listNotes.adapter = ArrayAdapter(this,
            android.R.layout.simple_list_item_1,
            DataManager.notes)
    }

}
